File: python/arrays/remove_duplicated.py
# Removes duplicates from a list and returns a new list with unique elements
# Time complexity: O(n), where n is the length of the input list

def remove_duplicates(nums):
    result = set(nums)  # O(n) to insert all elements into a set
    final = list(result)  # O(n) to convert set back to list
    return final  # O(1)

nums = [1, 1, 3, 3, 5, 6, 6]
print(remove_duplicates(nums))

# overall complexity = O(n) 

File: javascript/arrays/removeDuplicated.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var nums = [1, 1, 3, 3, 5, 6, 6];
var removeDuplicates = function (nums) {
  const result = new Set(nums); //loop nums inserting in a set = O(n)
  const final = Array.from(result); //convert set back to array - iteration = O(n)

  return final; //no effect on time O(1)
};

console.log(removeDuplicates(nums));

//overall complexity = O(n)
